### PR TITLE
Use chart plugins and remove code under visualizations

### PR DESCRIPTION
**Pull request by kristw**
### Changes in this PR 

This PR affects many files. 😱  However, the real edits only happen in 
- `package.json`
- `src/visualizations/presets/CommonChartPreset.js`
- `src/visualizations/presets/LegacyChartPreset.js`

The rest are deletions 💣. To sum up, here is what's happening. 

- Use chart plugins (formerly code under `visualizations` directory) that have been moved into `@superset-ui/legacy-*` repository and published as multiple `npm` packages. See https://apache-superset.github.io/superset-ui-legacy for demos of every chart and code at https://github.com/apache-superset/superset-ui-legacy
- Delete code under `visualizations` directory. 

Note: Currently applies to all charts except `TimeTable`, `FilterBox` and `deck.gl`.

### What does this mean for the project?

This PR enable charts that are plugins to be embeddable in another application using `ChartClient` and `<SuperChart />` from `@superset-ui/chart`. (Still need to work on demo and documentation. cc @xtinec)

This PR trim the core Superset codebase and make the chart plugins optional. The chart plugins potentially can be added/removed as the developers wish, or switched to different implementation, without everybody slamming all kinds of charts or customization into this repo, which makes the codebase become impossible to maintain over time and the Superset app has too many chart options that fits no one.  

### Future work

To enable developers to write their own chart plugins. There are still remaining work.

**Part 1.** Make control panels (left side of the explore page) part of the plugins

- Separate control panel config from `controls.jsx`
- Make control panel config part of chart plugins
- Migrate control panel configs to read from plugins

At this point developers will be enable to define a new chart and its control panel. However, the backend logic that would have been added to `viz.py` in the previous approach will be replaced by calling the new query api via `buildQuery` logic defined in the plugin. 

**Part 2.** We still have some work left on the backend to handle query objects created by `buildQuery`. Meanwhile the legacy charts will have `useLegacyApi` flag in the `ChartMetadata` to indicate that it is not using the new endpoint but use the old endpoints powered by `viz.py` instead.

**Part 3.** Documentation and better tooling for cross-repo work

- Example of independent plugin repo
- Generator for creating a new plugin repo 
- Handle symlink to plugin correctly when using `npm link` to enable interactive development.
- Update storybook in `@superset-ui/legacy-*` to preview live code. (Currently reading from the distribution code)

With that said, I think we have made a lot of progress in the past few months and this PR is one major step forward. There are still remaining work so please hold on if you are eager to add your own chart. In the meantime if the developers want to see previews of how to write a chart plugin, please look at 
https://github.com/apache-superset/superset-ui-legacy
and
https://github.com/apache-superset/superset-ui/packages/plugin-chart-word-cloud
for examples.

### CODE REVIEW

**Comment by mistercrunch** [comment]
destruction

**Comment by davidhassan** [comment]
−221,445 WOW!

**Comment by kristw** [comment]
Hi reviewers, I think this is good for a first pass. Please let me know if you have any concern. I may try to migrate the `MapBox` one as well but that should be pretty much the maximum scope of this PR. 
@williaster @mistercrunch @graceguo-supercat @michellethomas @xtinec

**Review by williaster — APPROVED** [review]
LGTM, thanks for all of your hard work on this and debugging the nvd3 issue! 

quite an epic PR 🙌 🏆 I can't believe the vis code had that many lines 😱

**Comment by joshbrooks** [comment]
@kristw I'm interested in whether/how this affects the "Country Map" - how should we now add a new country to visualisations?

File: superset/assets/src/visualizations/presets/CommonChartPreset.js [hunk 1]
```diff
@@ -17,21 +17,19 @@
  * under the License.
  */
 import { Preset } from '@superset-ui/core';
```
Comment (mgulaydin) [inline superset/assets/src/visualizations/presets/CommonChartPreset.js]: (senv) [root@blade02 presets]# pwd
/root/incubator-superset/superset/assets/src/visualizations/presets
(senv) [root@blade02 presets]# ll
total 8
-rw-r--r--. 1 root root 6453 Feb  4 16:33 MainPreset.js

Hello, there is no such a file in this directory

Note: superset version 0.35.2

**Pull request merged**